src/main/java/io/cryostat/discovery/DiscoveryStorage.java

```diff
@@ -260,10 +272,15 @@ public List<ServiceRef> listDiscoverableServices() {
     }
 
     public List<ServiceRef> listDiscoverableServices(String realm) {
```

Comment:
This method seems to list any services with the same name as the realm, and one place it seems to be in use is in the `CustomTargetPlatformClient`. Will there be a bug if someone names their plugin "Custom Targets"?

Reply:
True, if a plugin is registered that uses that same realm name then it could cause a conflict. I can get around this by providing the registration ID to the built-in `PlatformClient` implementations as a parameter to the `load` call.